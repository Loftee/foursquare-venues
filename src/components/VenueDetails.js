import React from 'react';

import './VenueDetails.scss';

function VenueDetails(props) {
    const {selectedVenue} = props;
    return (
        <div className={'venue-details__wrapper'}>
            <h2>Venue Details</h2>
            <div className={'venue-details'}>
                <h2 className={'venue-details__title'}>{selectedVenue.name}</h2>
                <p className={'venue-details__address'}>
                    {
                        selectedVenue.location.formattedAddress.join('\n')
                    }
                </p>
            </div>
        </div>
    )
}

export default VenueDetails;