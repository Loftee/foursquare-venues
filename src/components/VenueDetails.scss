.venue-details {
  border-radius: 5px;
  border: 1px solid white;
  padding: 15px;

  &__wrapper {
    float: left;
    width: 75%;
    padding: 0 15px;
  }

  &__address {
    white-space: pre-line;
  }

  @media (max-width: 767px) {
    margin-bottom: 50px;
  }
}