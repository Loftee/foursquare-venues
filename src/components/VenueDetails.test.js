import React from 'react';
import {shallow} from 'enzyme';
import VenueDetails from './VenueDetails';
import {testVenue} from "../testData";

describe('Venue Details', ()=> {
   const component = shallow(<VenueDetails selectedVenue={testVenue} />);

   it('renders correctly', () => {
      expect(component).toMatchSnapshot();
   });

   it('displays correct venue information', () => {
      expect(component.find('.venue-details__title').text()).toEqual('Costa Coffee');
      expect(component.find('.venue-details__address').text()).toEqual("4 Great Portland St\nLondon\nGreater London\nW1W 8QJ\nUnited Kingdom");
   });
});