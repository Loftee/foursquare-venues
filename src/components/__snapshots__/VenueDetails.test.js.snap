// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Venue Details renders correctly 1`] = `
<div
  className="venue-details__wrapper"
>
  <h2>
    Venue Details
  </h2>
  <div
    className="venue-details"
  >
    <h2
      className="venue-details__title"
    >
      Costa Coffee
    </h2>
    <p
      className="venue-details__address"
    >
      4 Great Portland St
London
Greater London
W1W 8QJ
United Kingdom
    </p>
  </div>
</div>
`;
